File: app/controller/status.go
package controller

import (
	"log"
	"net/http"
	"os"
	"time"

	"github.com/labstack/echo"
)

var StartTime time.Time
var hashesGenerated int
var keysGenerated int

func GetStatus(c echo.Context) error {
	name, err := os.Hostname()
	if err != nil {
		log.Println("Unable to get Hostname for status, error: ", err)
	}
	j, err := json.Marshal(map[string]interface{}{
		"status":          "OK",
		"uptime":          time.Since(StartTime).String(),
		"hostname":        name,
		"hashesGenerated": hashesGenerated,
		"keysGenerated":   keysGenerated,
	})
	if err != nil {
		return err
	}
	return c.JSONBlob(http.StatusOK, j)
}

File: src/status.rs
use serde::Serialize;
use std::time::{Duration, SystemTime};

#[derive(Serialize)]
pub struct Status {
    status: String,
    uptime: String,
    hostname: String,
    hashes_generated: usize,
    keys_generated: usize,
}

pub fn get_status() -> Status {
    let start_time = SystemTime::now();
    let uptime = match start_time.elapsed() {
        Ok(elapsed) => format!("{:?}", elapsed),
        Err(_) => "unknown".to_string(),
    };

    let hostname = match hostname::get() {
        Ok(name) => name.to_string_lossy().into_owned(),
        Err(_) => "unknown".to_string(),
    };

    Status {
        status: "OK".to_string(),
        uptime,
        hostname,
        hashes_generated: 0, // Replace with actual value
        keys_generated: 0,   // Replace with actual value
    }
}
